src/components/inline-rendered-diff.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+
+/**
+ * @typedef {Object} InlineRenderedDiffProps
+ * @property {Diff} diff The diff to render
+ * @property {Page} page The page this diff pertains to
+ */
+
+/**
+ * Display two versions of a page with their changes inline together.
+ *
+ * @class InlineRenderedDiff
+ * @extends {React.Component}
+ * @params {InlineRenderedDiffProps} props
+ */
+export default class InlineRenderedDiff extends React.Component {
+  constructor (props) {
+    super(props);
+    this.frame = null;
+  }
+
+  render () {
+    if (!this.props) {
+      return null;
+    }
+
+    return (
+      <div className="inline-render">
+        <iframe sandbox="allow-forms allow-scripts" ref={frame => this.frame = frame} />
+      </div>
+    );
+  }
+
+  /**
+     * @param {InlineRenderedDiffProps} nextProps
+     * @param {Object} nextState
+     * @returns {boolean}
+     */
+  shouldComponentUpdate (nextProps, nextState) {
+    return nextProps.diff.from_version_id !== this.props.diff.from_version_id
+      || nextProps.diff.to_version_id !== this.props.diff.to_version_id;
+  }
+
+  componentDidMount () {
+    this._updateContent();
+  }
+
+  componentDidUpdate () {
+    this._updateContent();
+  }
+
+  _updateContent () {
+    const raw_source = this.props.diff.content.diff;
+
+    this.frame.setAttribute(
+      'srcdoc',
+      renderableSource(raw_source, this.props.page));
+  }
+}
+
+/**
+ * Create renderable HTML source code from the raw HTML source of the diff.
+ *
+ * @param {string} source Full diff source code
+ * @param {Page} page The page that is being diffed
+ * @param {string} viewType Either `additions` or `removals`
+ */
+function renderableSource (source, page, viewType) {
```

Comment:
The `viewType` parameter is not used. Should we remove it or was it needed for something else?

Reply:
Nope, just a copy/paste error.